addons/mail/models/discuss/discuss_channel.py

```diff
@@ -764,6 +764,24 @@ def _message_subscribe(self, partner_ids=None, subtype_ids=None, customer_ids=No
         considered. """
         raise UserError(_('Adding followers on channels is not possible. Consider adding members instead.'))
 
+    def _get_access_action(self, access_uid=None, force_website=False):
+        """ Redirect to Discuss when possible instead of form view. """
+        self.ensure_one()
+        url = None
+        if self.env.user._is_internal():
+            url = f"/odoo/action-mail.action_discuss?active_id={self.id}"
+        elif self.channel_type != "chat":
+            url = f"/chat/{self.id}/{self.uuid}"
```

Comment:
I don't think we should be actively adding the `uuid`. Either the user has the right for the channel and the token is not necessary, or the user doesn't have access and then we should forbid it.